app/lib/clients/twitter_client.rb

```diff
@@ -1,17 +1,23 @@
 require 'twitter'
 
 class Clients::TwitterClient
-  def self.authenticate
-    client = Twitter::REST::Client.new do |config|
-      config.consumer_key = ENV['TWITTER_CONSUMER_KEY']
-      config.consumer_secret = ENV['TWITTER_CONSUMER_SECRET']
-      config.access_token = ENV['TWITTER_ACCESS_TOKEN']
-      config.access_token_secret = ENV['TWITTER_ACCESS_SECRET']
-    end
-    client
+
+  @client = Twitter::REST::Client.new do |config|
+    config.consumer_key = ENV['TWITTER_CONSUMER_KEY']
+    config.consumer_secret = ENV['TWITTER_CONSUMER_SECRET']
+    config.access_token = ENV['TWITTER_ACCESS_TOKEN']
+    config.access_token_secret = ENV['TWITTER_ACCESS_SECRET']
   end
 
-  TwitterQuery = Struct.new(:created_at, :text, :favorite_count, :retweet_count, :media, :link)
+  InteractionQuery = Struct.new(:favorite_count, :retweet_count)
+
+  MediaQuery = Struct.new(:image, :link)
+
+  TwitterQuery = Struct.new(:created_at, :text, :interactions, :media)
+
+  def self.media_query(latest)
+    MediaQuery.new(get_media(latest), get_link(latest))
```

Comment:
It may make more sense for latest to be an object that has these methods rather than class methods in client. Clients responsibility is to request and return the data. Another object could be in charge of how to resolve the data for the graphql query response